session.go

```diff
@@ -1557,8 +1619,8 @@ func (b *Batch) attempt(keyspace string, end, start time.Time, iter *Iter, host
 		End:        end,
 		// Rows not used in batch observations // TODO - might be able to support it when using BatchCAS
 		Host:    host,
+		Metrics: hostMetrics,
 		Err:     iter.err,
-		Attempt: b.attempts,
```

Comment:
If you're good with my suggestions above, you may want to restore this field and pass `b.totalAttempts`. On the other hand, query observers could be fine with calculating it from `hostMetrics`. Your call.